demo/dlb-init.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+NUM_VFS_PER_PF=${NUM_VFS_PER_PF:-1}
+
+bind() {
+    BSF=$1
+    DRIVER=$2
+
+    echo "$BSF" > /sys/bus/pci/drivers/"$DRIVER"/bind;
+}
+
+unbind() {
+    BSF=$1
+    OLD_DRIVER=$2
+
+    echo -n "$BSF" > /sys/bus/pci/drivers/"$OLD_DRIVER"/unbind;
+}
+
+#get dlb pf pci ids
+DEVS=$(realpath /sys/bus/pci/drivers/dlb2/????:??:??\.0)
+skip=true
+for dev in $DEVS; do
+  SRIOV_NUMVFS_PATH=$dev/sriov_numvfs
+  #every 2nd pf is used for configuring vfs
+  if $skip; then
+    skip=false
+    echo "$dev" "skipped"
+    continue
+  else
+    skip=true
+    echo "$dev" "configured"
+    if test -w "$SRIOV_NUMVFS_PATH" -a "$(cat "$SRIOV_NUMVFS_PATH")" -eq 0 ; then 
+      echo $((NUM_VFS_PER_PF)) | tee -a "$SRIOV_NUMVFS_PATH"
+    else
+      continue
+    fi    
+  fi
+
+  for N in $(seq 0 $((NUM_VFS_PER_PF-1)));do
+    #unbind vf
+    vf=$(basename "$(realpath "$dev"/virtfn"$N")")
+    unbind "$vf" dlb2
+
+    #configure vf
+    targets="num_atomic_inflights num_dir_credits num_dir_ports num_hist_list_entries num_ldb_credits num_ldb_ports num_ldb_queues num_sched_domains num_sn0_slots num_sn1_slots"
+    for target in $targets
+    do
+      val=$(cat "$dev"/total_resources/"$target")
+      share=$((val / NUM_VFS_PER_PF))
+      echo $share | tee -a "$dev"/vf"$N"_resources/"$target"
+    done
+
+    #bind back to dlb2 driver
+    bind "$vf" dlb2
+  done
```

Comment:
move this to a function that takes `$dev`  and the number of "VFs per PR"  as a parameter